### PR TITLE
arrow-cpp: 6.0.0 -> 6.0.1

### PR DESCRIPTION
**Pull request by cpcloud**
###### Motivation for this change

Bump `arrow-cpp` from 6.0.0 to 6.0.1

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by veprbl** [comment]
# Apache Arrow 6.0.1 (2021-11-10)

## Bug Fixes

* [ARROW-14437](https://issues.apache.org/jira/browse/ARROW-14437) - [Python] CSV test\_cancellation unittests fail on Apple M1
* [ARROW-14492](https://issues.apache.org/jira/browse/ARROW-14492) - [JS] Browser bundles broken
* [ARROW-14513](https://issues.apache.org/jira/browse/ARROW-14513) - [Release][Go] Update release-6.0.0 with /v6 suffix
* [ARROW-14519](https://issues.apache.org/jira/browse/ARROW-14519) - [C++] joins segfault when data contains list column
* [ARROW-14523](https://issues.apache.org/jira/browse/ARROW-14523) - [C++][Python] S3FileSystem write\_table can lose data
* [ARROW-14538](https://issues.apache.org/jira/browse/ARROW-14538) - [R] Work around empty tr call on Solaris
* [ARROW-14550](https://issues.apache.org/jira/browse/ARROW-14550) - [C++][Go] the json license is not compatible with FSF standards
* [ARROW-14583](https://issues.apache.org/jira/browse/ARROW-14583) - [R][C++] Crash when summarizing after filtering to no rows on partitioned data
* [ARROW-14584](https://issues.apache.org/jira/browse/ARROW-14584) - [Python][CI] Python sdist installation fails with latest setuptools 58.5
* [ARROW-14620](https://issues.apache.org/jira/browse/ARROW-14620) - [Python] Missing bindings for existing\_data\_behavior makes it impossible to maintain old behavior
* [ARROW-14630](https://issues.apache.org/jira/browse/ARROW-14630) - [C++] DCHECK in GroupByNode when error encountered


## New Features and Improvements

* [ARROW-13156](https://issues.apache.org/jira/browse/ARROW-13156) - [R] bindings for str\_count
* [ARROW-14181](https://issues.apache.org/jira/browse/ARROW-14181) - [C++][Compute] Hash Join support for dictionary	
* [ARROW-14189](https://issues.apache.org/jira/browse/ARROW-14189) - [Docs] Add version dropdown to the sphinx docs
* [ARROW-14310](https://issues.apache.org/jira/browse/ARROW-14310) - [R] Make expect\_dplyr\_equal() more intuitive
* [ARROW-14365](https://issues.apache.org/jira/browse/ARROW-14365) - [R] Update README example to reflect new capabilities
* [ARROW-14390](https://issues.apache.org/jira/browse/ARROW-14390) - [Packaging][Ubuntu] Add support for Ubuntu 21.10
* [ARROW-14433](https://issues.apache.org/jira/browse/ARROW-14433) - [Release][APT] Skip arm64 Ubuntu 21.04 verification
* [ARROW-14450](https://issues.apache.org/jira/browse/ARROW-14450) - [R] Old macos build error
* [ARROW-14459](https://issues.apache.org/jira/browse/ARROW-14459) - [Doc] Update the pinned sphinx version
* [ARROW-14480](https://issues.apache.org/jira/browse/ARROW-14480) - [R] Expose arrow::dataset::ExistingDataBehavior to R
* [ARROW-14486](https://issues.apache.org/jira/browse/ARROW-14486) - [Packaging][deb] libthrift-dev dependency is missing
* [ARROW-14490](https://issues.apache.org/jira/browse/ARROW-14490) - [Doc] Regenerate CHANGELOG.md to include all versions
* [ARROW-14496](https://issues.apache.org/jira/browse/ARROW-14496) - [Docs] Ensure links to non-sphinx parts of the docs are relative instead of absolute
* [ARROW-14499](https://issues.apache.org/jira/browse/ARROW-14499) - [Docs] Version dropdown side-by-side with search box
* [ARROW-14514](https://issues.apache.org/jira/browse/ARROW-14514) - [C++][R] UBSAN error on round kernel
* [ARROW-14580](https://issues.apache.org/jira/browse/ARROW-14580) - [Python] PyPI listing does not include trove classifiers for Python 3.10
* [ARROW-14623](https://issues.apache.org/jira/browse/ARROW-14623) - [Packaging][Java] Upload not only .jar but also .pom
* [ARROW-14628](https://issues.apache.org/jira/browse/ARROW-14628) - [Release][Python] Use python -m pytest

**Comment by veprbl** [comment]
@GrahamcOfBorg build python38Packages.pyarrow python39Packages.pyarrow

**Comment by cpcloud** [comment]
@veprbl I think the issue is jemalloc on aarch64-darwin. Not sure why though. Unfortunately I don't have access to a machine to test this on.

**Comment by veprbl** [comment]
> @veprbl I think the issue is jemalloc on aarch64-darwin. Not sure why though. Unfortunately I don't have access to a machine to test this on.

It's already failing on Hydra https://hydra.nixos.org/build/158150169 . I would leave it be for now – it may resolve by itself.

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 146823` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.spacy-transformers</li>
    <li>python39Packages.spacy-transformers</li>
  </ul>
</details>
<details>
  <summary>6 packages failed to build:</summary>
  <ul>
    <li>python38Packages.intake</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>arrow-cpp</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.intake-parquet</li>
    <li>python38Packages.pyarrow</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.intake-parquet</li>
    <li>python39Packages.pyarrow</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
  </ul>
</details>

**Review by veprbl — APPROVED** [review]

**Comment by cpcloud** [comment]
@veprbl Thanks for fixing. Now that parquet-tools is merged, I'll probably need to adjust the tests. Looking into that now.

File: pkgs/tools/misc/parquet-tools/default.nix [hunk 1]
```diff
@@ -32,7 +32,7 @@ buildPythonApplication rec {
       --replace 'thrift = "^0.13.0"' 'thrift = "*"' \
       --replace 'halo = "^0.0.29"' 'halo = "*"'
     substituteInPlace tests/test_inspect.py \
-      --replace "parquet-cpp-arrow version 5.0.0" "parquet-cpp-arrow version 6.0.0" \
+      --replace "parquet-cpp-arrow version 5.0.0" "parquet-cpp-arrow version 6.0.1" \
```
Comment (veprbl) [inline pkgs/tools/misc/parquet-tools/default.nix:35]:
```suggestion
      --replace "parquet-cpp-arrow version 5.0.0" "parquet-cpp-arrow version ${arrow-cpp.version}" \
```

**Comment by cpcloud** [comment]
@veprbl Is this ready to merge?

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 146823` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.spacy-transformers</li>
    <li>python39Packages.spacy-transformers</li>
  </ul>
</details>
<details>
  <summary>7 packages failed to build:</summary>
  <ul>
    <li>python38Packages.google-cloud-bigquery</li>
    <li>python38Packages.intake</li>
    <li>python38Packages.tokenizers</li>
    <li>python38Packages.transformers</li>
    <li>python39Packages.intake</li>
    <li>python39Packages.tokenizers</li>
    <li>python39Packages.transformers</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>arrow-cpp (parquet-cpp)</li>
    <li>parquet-tools</li>
    <li>python38Packages.datasets</li>
    <li>python38Packages.intake-parquet</li>
    <li>python38Packages.pyarrow</li>
    <li>python38Packages.snowflake-connector-python</li>
    <li>python38Packages.snowflake-sqlalchemy</li>
    <li>python39Packages.datasets</li>
    <li>python39Packages.google-cloud-bigquery</li>
    <li>python39Packages.intake-parquet</li>
    <li>python39Packages.pyarrow</li>
    <li>python39Packages.snowflake-connector-python</li>
    <li>python39Packages.snowflake-sqlalchemy</li>
  </ul>
</details>

**Pull request merged**